### PR TITLE
test(core): add failing test for path heap-buffer-overflow in metacall_load_from_file

### PR DESCRIPTION
**Pull request by anmoltrivedi2312-ux**
Hi @viferga,

Opening this PR with the failing reproduction test as per the TDD plan.

In `metacall_load_from_file_ex` (in `metacall.c`), paths are copied into `loader_path` buffers with `strncpy(..., LOADER_PATH_SIZE)`, which leaves the buffer unterminated on inputs of 4096 bytes or more. This leads to `memcpy` in `loader_impl_handle_name()` over-reading by one byte from a 4096-byte heap allocation, which fires an AddressSanitizer `heap-buffer-overflow`.

The test is in `source/tests/metacall_path_overflow_test` and runs against the built-in `mock` loader, and thus does not require any additional language runtimes to be installed.

Once CI detects the ASan crash, I'll push the fix commit to this PR directly to turn it green. 
---
### AI Assistance Disclosure
An LLM was used to assist in identifying this issue. Below is the system prompt and audit framework used for the analysis:

<details>
<summary>Click to view full System Prompt</summary>

**Role and Objective**
You are a Principal Systems Engineer and Polyglot Runtime Specialist auditing open-source codebases for the MetaCall ecosystem (specifically metacall/core, metacall/ssr, and satellite plugins).
Your goal is to inspect the codebase, identify legitimate, non-trivial bugs, performance bottlenecks, edge-case failures, and architectural regressions, and draft actionable GitHub issue reports.

**Audit Scope & Domain Checklists**
1. Core Runtime Focus (metacall/core - C / C++ / FFI)
* Memory & Lifecycle: Unchecked allocations, dangling pointers, double-frees across FFI boundaries. Mismatched memory owners. Missing cleanup in failure/early-exit branches.
* Cross-Runtime Synchronization & Concurrency: Thread-safety violations when invoking foreign functions across distinct runtime threads. Improper acquisition or release of runtime locks. Race conditions in plugin loading.
* Error Handling & Type Marshaling: Silent truncation or undefined behavior in dynamic type casting. Unhandled native exceptions or panics.
* Platform & Portability: Non-standard compiler extensions, architecture-specific pointer assumptions, or OS-specific dynamic linking quirks.

2. SSR Focus (metacall/ssr - TypeScript / Node / Polyglot Web)
* Resource Retention & Memory Leaks: Unbounded cache growth, event listener leaks. Worker thread/process lifecycle leaks.
* Stream Handling & Async Integrity: Backpressure mismanagement, unhandled promise rejections, incomplete request abort signal handling.
* Security & Input Sanitization: Unsanitized data serialization passed from foreign runtime contexts into rendered HTML.

**Audit Methodology**
Static Analysis & Code Path Tracing: Trace execution paths starting from public API surfaces down to internal runtime hooks.
False-Positive Elimination: Verify if the condition is already guarded earlier. Check standard runtime guarantees.
Impact Assessment: Prioritize issues that lead to crashes, memory leaks, silent data corruption, or broken cross-language contract guarantees.

</details>

### CODE REVIEW

File: source/loader/source/loader_impl.c [hunk 1]
```diff
@@ -791,11 +791,11 @@ int loader_impl_initialize(plugin_manager manager, plugin p, loader_impl impl)
 	library_path = plugin_manager_library_path(manager);
 
 	/* The library path priority order is the following:
-	* 1) In the configuration <tag>_loader.json, the flag loader_library_path.
-	* 2) The environment variable LOADER_LIBRARY_PATH.
-	* 3) The directory of the path of (lib)metacall.{so,dll,dylib,...}
-	* 4) Default install paths defined at compile time.
-	*/
+	 * 1) In the configuration <tag>_loader.json, the flag loader_library_path.
```
Comment (viferga) [inline source/loader/source/loader_impl.c]: please undo all this, it's very annoying, only push the effective code

File: source/loader/source/loader_impl.c [hunk 2]
```diff
@@ -317,7 +317,7 @@ void loader_impl_configuration_environment(loader_impl impl)
 	 *     ...
 	 * ]
 	 * Reference: https://github.com/metacall/core/issues/760
-	*/
+	 */
```
Comment (viferga) [inline source/loader/source/loader_impl.c]: same here, avoid always pushing that kind of style changes...

**Comment by anmoltrivedi2312-ux** [comment]
So sorry about that! It was an auto-formatter which was messing with untouched comments in my editor (I've disabled the extension but reverted all the formatting noises) and force-pushed. Now, the diff strictly contains the functional fix (+11 / -0 in both files).

File: source/loader/source/loader_impl.c [hunk 3]
```diff
@@ -1351,6 +1351,10 @@ size_t loader_impl_handle_name(plugin_manager manager, const loader_path path, l
 	}
 
 	size_t length = strnlen(path, LOADER_PATH_SIZE);
+	if (length >= LOADER_PATH_SIZE)
```
Comment (viferga) [inline source/loader/source/loader_impl.c]: I think this is unnecessary.
